### PR TITLE
Zendesk: redirect external users to generic support page

### PR DESCRIPTION
**Pull request by nkiruka**
# Description
**SWITCH BASE BRANCH TO STAGING PRIOR TO MERGING

External users with non code.org email addresses are redirected to support.code.org when they try to sign in.

This PR is part of the [epic](https://codedotorg.atlassian.net/browse/FND-685).   The PR will be merged after [PR 30642](https://github.com/code-dot-org/code-dot-org/pull/30642) is merged.

**Note, if the base branch is switched to zendesk-sso, the specific changes relevant to this PR are not shown because of a merge conflict that was resolved (staging was merged due to a change in config.yml)

- [jira](https://codedotorg.atlassian.net/browse/FND-799)

## Testing story

  Does your change include appropriate tests? 
   Yes, a unit test was added.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by sureshc** [comment]
I merged the latest from the `staging` branch into the `zendesk-sso` branch.  You should set `zendesk-sso` as the base branch of this Pull Request, since this change is dependent on `zendesk-sso`.

**Comment by joshlory** [comment]
Will that create the right diff? Other option would be to merge `zendesk-sso` -> `staging` (PR link?) then `staging` -> this branch.

**Comment by nkiruka** [comment]
@joshlory I am not sure why initially when I set the base branch as "zendesk-sso", it showed ~900 commits, which I attributed to the commits from merging staging.

The base branch is now set to "zendesk-sso" and shows the correct diff for the changes made.

**Comment by dju90** [comment]
We don't need to block this PR on #30642 since this only applies to external users, correct?

File: dashboard/app/controllers/zendesk_session_controller.rb
Comment (islemaster) [inline dashboard/app/controllers/zendesk_session_controller.rb:12]: I don't think you can initialize an instance variable right in the class definition like this - it may need to go inside a `def initialize` block.  On the other hand, in Ruby it might be totally unnecessary to initialize an instance variable to `nil`, so you might be able to just drop this line.

File: dashboard/app/controllers/zendesk_session_controller.rb
Comment (islemaster) [inline dashboard/app/controllers/zendesk_session_controller.rb:31]: Should this be `user.email.present?`, as below?

**Review by islemaster — COMMENTED** [review]
A couple of comments to address, otherwise this is looking good.  I'm also totally unfamiliar with this code - I'd recommend also getting a review from someone on Foundations.

**Review by islemaster — APPROVED** [review]
:+1: Approved after reviewing your specific commits here.  Thanks!

**Comment by bethanyaconnor** [comment]
Do we still want this with Zendesk now being a direct SSO?

**Comment by dju90** [comment]
Touched base with Josh and Nkiru on this, and yes, unfortunately the new direct SSO to Zendesk makes this change unnecessary

**Comment by nkiruka** [comment]
zendesk controller to be removed.

**Pull request closed without merging**
